### DISCUSSION TITLE
Problem with image upload / cloudinary

### DISCUSSION THREAD

**Discussion by janneserichsen** [neuefische/questions#527] [Web]
Hello Yall i have a Problem with my Image Upload ive done it like in the Workshop to it and i get this Error now in the Console
 Uncaught (in promise) SyntaxError: Unexpected token '<', "<!DOCTYPE "... is not valid JSON

<img width="403" alt="Bildschirmfoto 2023-02-13 um 11 54 44" src="https://user-images.githubusercontent.com/118190990/218440842-bdfd04eb-ab68-46e0-a563-ec2f54c3e6ed.png">

<img width="481" alt="Bildschirmfoto 2023-02-13 um 11 54 22" src="https://user-images.githubusercontent.com/118190990/218440527-024091b3-ca3d-4423-8b67-f12b5286c97c.png">
-23b755ef2ba3.png">

**_[REDACTED]_**

<img width="563" alt="Bildschirmfoto 2023-02-13 um 11 55 36" src="https://user-images.githubusercontent.com/118190990/218441173-b1e769f4-39e5-4ce6-a5e8-9ef1698bcea1.png">
<img width="729" alt="Bildschirmfoto 2023-02-13 um 11 55 28" src="https://user-images.githubusercontent.com/118190990/218441368-5933f034-41ce-4183-a513-90e638d8e99e.png">

Can someone please help me with this one

**Comment by shebtastic** [comment]
Hey!

`Uncaught (in promise) SyntaxError: Unexpected token '<', "<!DOCTYPE "... is not valid JSON` is usually a sign, that there is a hard error happening on the backend. That usually comes down to environment variables to not have been set.

Is the issue happening on your machine locally or on your vercel deployment?
Your environment variables need to be either set using the `export` command on your shell session or using `.env`-files.

If you need further help, could you post a PR / link to commit of your code?

**Reply by janneserichsen** [reply]
Hey first thing’s first thanks for your anwer
 I tried both of your tipps but none worked so heres the link to my pr
https://github.com/janneserichsen/CapStone-Graffstar/pull/13/files

**Reply by shebtastic** [reply]
if the `.env.example` looks similar to your actual `.env`, you seem to have a syntax error there.

https://nextjs.org/docs/basic-features/environment-variables#loading-environment-variables

file should have a syntax like this: 

```.env
CLOUD_NAME=
API_KEY=
API_SECRET=
```

with the actual values on the right hand side of the `=` without any spaces in between.
Make sure _not to commit(!)_ your actual secrets, otherwise you leak your credentials and they are part of your repositories history.

**Reply by janneserichsen** [reply]
I tried it and its still not working 

<img width="138" alt="Bildschirmfoto 2023-02-13 um 15 07 08" src="https://user-images.githubusercontent.com/118190990/218479915-2eb87034-9a3a-4c2f-a08d-981f063fcad0.png">

**Reply by shebtastic** [reply]
The file still looks malformed. You should not have any empty spaces where they don't belong.
I tested it locally and the upload works with a correctly defined `.env`-file.

Rendering the image will not work, unless you configure the `next.config.js`.
